spring-core/src/main/java/org/springframework/util/StringUtils.java

```diff
@@ -774,7 +774,7 @@ public static String uriDecode(String source, Charset charset) {
 				bos.write(ch);
 			}
 		}
-		return (changed ? new String(bos.toByteArray(), charset) : source);
+		return changed ? StreamUtils.baosToString(bos, charset) : source;
```

Comment:
Please note that we typically leave the enclosing parentheses intact for such ternary statements.

Reply:
Reverted